### ISSUE TITLE
Error while running embed step in Web UI

### ISSUE THREAD

**Issue by dhruv-anand-aintech** [enjalot/latent-scope#30]
Hi there, I was running the following step in the UI:

Running embed
ls-embed info_map input_prompt transformers-jinaai___jina-embeddings-v2-small-en --prefix=""

And I got this error:
```
Loading environment variables from: /Users/dhruvanand/Code/vector-io/.env
RUNNING: embedding-001
loading jinaai/jina-embeddings-v2-small-en

tokenizer_config.json:   0%|          | 0.00/373 [00:00<?, ?B/s]
tokenizer_config.json: 100%|██████████| 373/373 [00:00<00:00, 309kB/s]

vocab.txt:   0%|          | 0.00/232k [00:00<?, ?B/s]
vocab.txt: 100%|██████████| 232k/232k [00:00<00:00, 606kB/s]
vocab.txt: 100%|██████████| 232k/232k [00:00<00:00, 602kB/s]

tokenizer.json:   0%|          | 0.00/712k [00:00<?, ?B/s]
tokenizer.json: 100%|██████████| 712k/712k [00:00<00:00, 948kB/s]
tokenizer.json: 100%|██████████| 712k/712k [00:00<00:00, 945kB/s]

special_tokens_map.json:   0%|          | 0.00/125 [00:00<?, ?B/s]
special_tokens_map.json: 100%|██████████| 125/125 [00:00<00:00, 229kB/s]

config.json:   0%|          | 0.00/1.18k [00:00<?, ?B/s]
config.json: 100%|██████████| 1.18k/1.18k [00:00<00:00, 3.51MB/s]
A new version of the following files was downloaded from https://huggingface.co/jinaai/jina-bert-implementation:
- configuration_bert.py
. Make sure to double-check they do not contain any added malicious code. To avoid downloading new versions of the code file, you can pin a revision.

modeling_bert.py:   0%|          | 0.00/97.7k [00:00<?, ?B/s]
modeling_bert.py: 100%|██████████| 97.7k/97.7k [00:00<00:00, 480kB/s]
modeling_bert.py: 100%|██████████| 97.7k/97.7k [00:00<00:00, 477kB/s]
A new version of the following files was downloaded from https://huggingface.co/jinaai/jina-bert-implementation:
- modeling_bert.py
. Make sure to double-check they do not contain any added malicious code. To avoid downloading new versions of the code file, you can pin a revision.

model.safetensors:   0%|          | 0.00/65.4M [00:00<?, ?B/s]
model.safetensors:  16%|█▌        | 10.5M/65.4M [00:01<00:08, 6.37MB/s]
model.safetensors:  32%|███▏      | 21.0M/65.4M [00:02<00:04, 10.7MB/s]
model.safetensors:  48%|████▊     | 31.5M/65.4M [00:02<00:02, 13.6MB/s]
model.safetensors:  64%|██████▍   | 41.9M/65.4M [00:03<00:01, 15.3MB/s]
model.safetensors:  80%|████████  | 52.4M/65.4M [00:03<00:00, 16.4MB/s]
model.safetensors:  96%|█████████▌| 62.9M/65.4M [00:04<00:00, 17.3MB/s]
model.safetensors: 100%|██████████| 65.4M/65.4M [00:04<00:00, 17.4MB/s]
model.safetensors: 100%|██████████| 65.4M/65.4M [00:04<00:00, 14.6MB/s]
Checking for empty inputs
embedding 132 sentences in 1 batches

0%|          | 0/1 [00:00<?, ?it/s]/Users/dhruvanand/miniforge3/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
warnings.warn('resource_tracker: There appear to be %d '

```

No embeddings got produced. Could you help debug what caused this?

**Comment by enjalot** [comment]
`UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
warnings.warn('resource_tracker: There appear to be %d '`

I've encountered this a few times when it seems like I run out of memory running a model (happened to me more often when using a chat model)

honestly I don't know how to catch the error or prevent it 😞
the subprocess doesn't seem to exit in a way that the parent process catches like it would if it ran into some other kind of error

**Comment by dhruv-anand-aintech** [comment]
I'm on an M1 Pro. Jina small works fine with batch size 1 using a python notebook.

**Comment by enjalot** [comment]
I've added a new option to the web UI (and the scripts) that lets you change the batch size when embedding. please try it out and let me know if you still have the issue

**Comment by dhruv-anand-aintech** [comment]
Can't see it 
![Screenshot 2024-03-06 at 2 13 22 PM](https://github.com/enjalot/latent-scope/assets/105786647/7d886a5e-8c86-4cb3-a454-56cb81e51647)

**Comment by enjalot** [comment]
@dhruv-anand-aintech yes sorry about that. I found a problem in my packaging process that meant the web ui code was not being consistently bundled when packaging. I've made a more robust script to build the package and published 0.1.5 which should now have the proper batch size in the UI

thanks for helping me sand all these rough edges!

**Comment by dhruv-anand-aintech** [comment]
sure thing! I can imagine the struggle, having dealt with the same recently with Vector-io :)
